.github/workflows/build.yml

```
@@ -237,18 +266,23 @@ jobs:
       - name: Prepare list of release files & release notes
         id: makeinfo
         run: |
+          if ! ls ${{ env.base_name }}.txt > /dev/null 2>&1; then
```

```suggestion
          if ! [[ -e ${{ env.base_name }}.txt ]]; then
```
Removed the useless `ls`.

---

This won't work, because env.base_name is a glob pattern, not a single file.

---

Oh, maybe this instead:
```suggestion
          files=( ${{ env.base_name }}.txt )
          if [[ ! -e ${files[0]} ]]; then
```
or alternatively:
```suggestion
          for file in ${{ env.base_name }}.txt; do
              if [[ ! -e $file ]]; then
                  # ...
              fi
              break
           done
```
Both would remove the usage of `ls`.

---

Why do you think getting rid of `ls` is a good idea?
Thinking of bash specific way - using `compgen -G` to test if glob pattern matches anything might be better than expanding and testing individual file probably.

---

Yes, I was suggesting to remove the usage of unnecessary external commands, although `ls` could also fail for reasons unrelated to the existence of the files. Both of the above suggestions would only test the first file (note the `break` in the second example), which would be needed to check if the glob expanded to a least one actual file. Instead, one could enable `nullglob` and then check if any files were matched.

I have not used the `compgen -G` built-in myself, although it looks like you would need to put the argument in quotes to delay the expansion until after it is called.

---

You're right about the quotes, thanks!